packages/model-viewer/src/features/ar.ts

```diff
@@ -432,14 +432,35 @@ configuration or device capabilities');
 
       updateSourceProgress(0.2);
 
-      const exporter = new USDZExporter();
-      const arraybuffer = await exporter.parse(model);
+      const usdzOptions = {
+        ar: {
+          anchoring: { type: 'plane' },
+          planeAnchoring: {
+            alignment: this.arPlacement === 'wall' ? 'vertical' : 'horizontal',
+          },
+        },
+      };
+
+      // necessary because quicklook internally rotates model when placing vertical. 
+      // See https://github.com/google/model-viewer/issues/3989
+      if (usdzOptions.ar.planeAnchoring.alignment === 'vertical') {
+        model.rotateX((-90 * Math.PI) / 180);
```

Comment:
Let's not update the GLB; it's a good test precisely because many models are not aligned to the origin. I believe you'll need to also apply the `scene.target` translation to make it work.

Reply:
It sounds like USDZ may have particular assumptions about where the origin should be relative to the placement point, so you may also need to make use of `$scene.boundingBox` to find the translation you need.

Reply:
Would you like to take over this PR or make your own version?

Reply:
I didn't understand what you were suggesting in the comments above with the translation and bounding box, and also every time I tried to do threejs operation like that translation followed by rotation I always do things wrong and lose my hairs, so I probably can't do any change related to that.
But not sure if we should go this way if removing the lines https://github.com/mrdoob/three.js/blob/aaafb4ffe8830932e31aa44003376457265d126a/examples/jsm/exporters/USDZExporter.js#L209-L210 also fixes the issue.

Reply:
related comment https://github.com/google/model-viewer/issues/3989#issuecomment-2030217299